packages/go_router/lib/src/state.dart

```diff
@@ -157,9 +157,13 @@ class GoRouterState {
     String name, {
     Map<String, String> pathParameters = const <String, String>{},
     Map<String, String> queryParameters = const <String, String>{},
+    String? fragment,
   }) {
+    // Generate base location using configuration, with optional path and query parameters
+    // Then conditionally append fragment if it exists and is not empty
     return _configuration.namedLocation(name,
-        pathParameters: pathParameters, queryParameters: queryParameters);
+            pathParameters: pathParameters, queryParameters: queryParameters) +
+        ((fragment?.isNotEmpty ?? false) ? '#$fragment' : '');
```

Comment:
we should add fragment to namedLocation api as well

Reply:
Sorry for overlooking that part . Added the the fragment handling to the namedLocation api.